cpp/include/raft/linalg/gemm.cuh

```diff
@@ -27,8 +28,65 @@
 #include <raft/core/resources.hpp>
 #include <raft/util/input_validation.hpp>
 
-namespace raft {
-namespace linalg {
+namespace raft::linalg {
+
+/**
+ * @brief the wrapper of cublasLt matmul function
+ *  It computes the following equation: C = alpha .* opA(A) * opB(B) + beta .* C
+ *
+ * @tparam DevicePointerMode whether pointers alpha, beta point to device memory
+ * @tparam S the type of scale parameters alpha, beta
+ * @tparam A the element type of matrix A
+ * @tparam B the element type of matrix B
+ * @tparam C the element type of matrix C
+ *
+ * @param [in] res raft resources
+ * @param [in] trans_a cublas transpose op for A
+ * @param [in] trans_b cublas transpose op for B
+ * @param [in] m number of rows of C
+ * @param [in] n number of columns of C
+ * @param [in] k number of rows of opB(B) / number of columns of opA(A)
+ * @param [in] alpha host or device scalar
+ * @param [in] a_ptr such a matrix that the shape of column-major opA(A) is [m, k]
+ * @param [in] lda leading dimension of A
+ * @param [in] b_ptr such a matrix that the shape of column-major opA(B) is [k, n]
+ * @param [in] ldb leading dimension of B
+ * @param [in] beta host or device scalar
+ * @param [inout] c_ptr column-major matrix of size [m, n]
+ * @param [in] ldc leading dimension of C
+ */
+template <bool DevicePointerMode = false, typename S, typename A, typename B, typename C>
+void matmul(raft::resources const& res,
+            bool trans_a,
+            bool trans_b,
+            uint64_t m,
+            uint64_t n,
+            uint64_t k,
+            const S* alpha,
```

Comment:
Pointer-based APIs are deprecated. Please use mdspan (or at least add an mdspan variant to the group that will be included in the docs below and test it). Eventually this one will be removed with the other pointer-based APIs.

Reply:
resoling: I've removed this altogether, because it would mirror the mdspan api of the gemm that is already defined in this file.